pkgs/build-support/fetchurl/mirrors.nix

```diff
@@ -405,4 +405,12 @@ rec {
     http://repo1.maven.org/maven2/
     http://central.maven.org/maven2/
   ];
+
+  # Alsa Project
+  alsa = [
+     ftp://ftp.alsa-project.org/pub/
+     http://alsa.cybermirror.org/
```

Comment:
For what it's worth, this one is not listed on the upstream project page:
http://alsa-project.org/main/index.php/Download

and it appears to be "temporarily down for maintenance and will be back shortly":
http://alsa.cybermirror.org/

It's also worth noting that many enterprise firewalls and system configurations are generally more suspicious / less tolerant of FTP traffic than HTTP traffic, and may block it entirely for security considerations.  Whenever possible it's nice to provide some HTTP alternatives!